hibernate-core/src/main/java/org/hibernate/sql/results/graph/instantiation/internal/InstantiationHelper.java

```diff
@@ -56,13 +56,31 @@ private static boolean checkArgument(Class<?> targetJavaType, BeanInfo beanInfo,
 
 	public static boolean isConstructorCompatible(Class<?> javaClass, List<Class<?>> argTypes, TypeConfiguration typeConfiguration) {
 		for ( Constructor<?> constructor : javaClass.getDeclaredConstructors() ) {
-			if ( isConstructorCompatible( constructor, argTypes, typeConfiguration) ) {
+			if ( isConstructorCompatible( constructor, argTypes, typeConfiguration ) ) {
 				return true;
 			}
 		}
 		return false;
 	}
 
+	public static <T> Constructor<T> findMatchingConstructor(
+			Class<T> type,
+			List<Class<?>> argumentTypes,
+			TypeConfiguration typeConfiguration) {
+		try {
+			return type.getDeclaredConstructor( argumentTypes.toArray( new Class[0] ) );
```

Comment:
I would just always scan through the list of declared constructors. I wouldn't expect there to be a lot of constructors anyway and if you look at the implementation of `getDeclaredConstructor` you will notice that it scans through the declared constructors as well :)

Reply:
Sounds good, thanks!